### PR TITLE
chore: Document field error on S3 Bucket types

### PR DESCRIPTION
**Pull request by achetronic**
### Description of your changes

Just add a bit of documentation in the field locationConstraint for S3 buckets, to help people to debug some "error dialing" given by AWS and not by the provider

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by haarchri** [comment]
Can you run make generate locally ?

**Comment by achetronic** [comment]
> Can you run make generate locally ?

Yes! :) Sorry, I forgot. Manifest generated and added to the PR

File: apis/s3/v1beta1/bucket_types.go [hunk 1]
```diff
@@ -35,6 +35,9 @@ type BucketParameters struct {
 
 	// LocationConstraint specifies the Region where the bucket will be created.
 	// It is a required field.
+	// Due to AWS API limitations lacking on a proper response, when this field is set to a wrong value,
+	// or to non-existent region on bucket creation, it's impossible forwarding a meaning status message to the user
+	// about the problem, producing some errors related to dial
```
Comment (MisterMX) [inline apis/s3/v1beta1/bucket_types.go:40]:
```suggestion
	// Due to AWS API limitations lacking on a proper response, when this field is set to a wrong value,
	// or to non-existent region on bucket creation, it's impossible forwarding a meaningful status message to the user
	// about the problem, producing connection errors instead.
```

**Review by MisterMX — CHANGES_REQUESTED** [review]

**Comment by achetronic** [comment]
> LGTM. Thanks @achetronic!

Thank you and @haarchri for guiding me on this :)

**Pull request merged**